pkg/cloud/azure/actuators/machine/reconciler.go

```diff
@@ -188,21 +188,16 @@ func (s *Reconciler) Update(ctx context.Context) error {
 	}
 
 	if vm.NetworkProfile != nil && vm.NetworkProfile.NetworkInterfaces != nil {
-		var vnetName string
 		if s.scope.MachineConfig.Vnet == "" {
```

Comment:
nit, we should check all this expectations in a `validate()` func and early fail